File: solutions/python/02/abc142_b/main.py
import sys


def main() -> None:
    input = sys.stdin.readline
    n = int(input())
    a = sorted(map(int, input().split()))
    k = 2
    total = 0
    for i, price in enumerate(a):
        remaining = n - i
        if remaining % k == 0:
            total += price // 2
        else:
            total += price
    print(total)


if __name__ == "__main__":
    main()

File: solutions/go/02/abc142_b/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	var n int
	fmt.Fscan(in, &n)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(in, &a[i])
	}
	sort.Ints(a)
	const k = 2
	total := 0
	for i, price := range a {
		remaining := n - i
		if remaining%k == 0 {
			total += price / 2
		} else {
			total += price
		}
	}
	fmt.Println(total)
}
